[docs] Split the Editing page

cherniavskii opened the pull request:
Part of https://github.com/mui/mui-x/issues/7762

Changes:
- split Editing page into multiple pages
- move the editing recipes page to Editing -> Examples page

Preview: https://deploy-preview-7772--material-ui-x.netlify.app/x/react-data-grid/editing/

alexfauquette commented on docs/.link-check-errors.txt:
```diff
@@ -2,5 +2,6 @@ Broken links found by `yarn docs:link-check` that exist:
 
 - https://mui.com/blog/material-ui-v4-is-out/#premium-themes-store-✨
 - https://mui.com/size-snapshot
+- https://mui.com/x/react-data-grid/editing/#row-editing
```
This will require an update of the material-ui repo. Especially the `PricingTable.tsx`

cherniavskii replied:
Yes, after releasing the v6 stable

alexfauquette reviewed (approved):
Naming sounds nice.  👌

I did not read entirely the page content assuming it's copied past

I'm wondering, do you plan to add sections in "row editing" page. Otherwise, it could go to the overview page with 
- `editMode="row"` section between "making colums editable" and "value parser/setter"
- CRUD example at then end

For the two other sections, they are shorts but very advanced and well-scoped, so not a problem

cherniavskii commented:
> I'm wondering, do you plan to add sections in "row editing" page. Otherwise, it could go to the overview page with
> - editMode="row" section between "making colums editable" and "value parser/setter"
> - CRUD example at then end

The CRUD example is part of the Row editing section, so I would rather not split them.

LukasTy commented on docs/data/pages.ts:
```diff
@@ -88,10 +98,7 @@ const pages: MuiPage[] = [
-          { pathname: '/x/react-data-grid/recipes-editing', title: 'Editing' },
-          { pathname: '/x/react-data-grid/recipes-row-grouping', title: 'Row grouping' },
-        ],
+        children: [{ pathname: '/x/react-data-grid/recipes-row-grouping', title: 'Row grouping' }],
```
What would be the plan with this page/group?
- Rename to `Examples`;
- Refactor in a similar way with a follow-up PR and remove;

cherniavskii replied:
My plan regarding this page:
- Remove the `Group & Pivot` group and move its pages one level up. This group seems a bit artificial anyway - for example, Aggregation doesn't fit this group. Also, the features in this group are hard to discover (as opposed to other features one level above).
- Split Row grouping into multiple pages and group them, then move the grouping recipes to this group

What do you think?

LukasTy replied:
I do understand and agree with the first point. 👌 Really great points regarding `Group & Pivot`.
As far as the second point goes, I feel a bit lost with your provided description. 🤔 
Would the final result be, that the `Recipes` section would be removed and what is currently there would end up under `Row grouping -> Examples`?
Not to mention the other new sections/groups extracted from `Group & Pivot`.
If that's the plan, it does sound very nice to me. 👍 ✅

cherniavskii replied:
> Would the final result be, that the Recipes section would be removed and what is currently there would end up under Row grouping -> Examples?

Yes. Although we might add a top-level Recipes (or Examples) page for cross-feature examples that we cannot clearly categorize.

For the other pages from the Group & Pivot - we might split some of them too. I think Aggregation might be a good candidate for the page split

LukasTy replied:
Thanks for the clarification. 🙏

LukasTy reviewed (approved):
Looks great! 💯 
A similar makeover (splitting into several smaller pages) has been done to the pickers getting started page quite successfully recently. 👍

MBilalShafi commented:
@cherniavskii Would you like to proceed with it or do you see a problem? I am trying to do a similar division of pages for Filtering.

cherniavskii commented:
I've updated the PR and split the editing page again, here's the current structure: https://deploy-preview-7772--material-ui-x.netlify.app/x/react-data-grid/editing/

<img width="261" src="https://github.com/mui/mui-x/assets/13808724/426b92f0-55d4-43f7-8fea-9d83154e97af">

cherniavskii commented on docs/data/pages.ts:
```diff
@@ -65,7 +65,17 @@ const pages: MuiPage[] = [
+          { pathname: '/x/react-data-grid/editing/persistence' },
+          { pathname: '/x/react-data-grid/editing/custom-edit-component' },
+          { pathname: '/x/react-data-grid/editing/editing-events' },
+          { pathname: '/x/react-data-grid/editing/recipes' },
```
We didn't reach an agreement on "Examples" vs "Recipes" in https://github.com/mui/mui-x/issues/7762 😅

I kept the previous title ("Recipes") for now, feel free to vote on this comment if you have a preference:
- 🚀 for "Recipes"
- 🎉 for "Examples"

cherniavskii replied:
@richbustos What do you think about this one?

richbustos reviewed (approved):
LGTM!

MBilalShafi commented on docs/data/data-grid/editing/custom-edit-component.md:
```diff
@@ -0,0 +1,126 @@
+
+<p class="description">Creating custom edit component.</p>
+
+## Create your own edit component
```
Could we discard this heading in favor of the main one?

MBilalShafi reviewed (approved):
Nice update

joserodolfofreitas reviewed (approved):
Bumping this awesome update.
